app.js

```diff
@@ -55,6 +60,23 @@ app.use(function (err, req, res, next) {
   res.status(500).send('Something broke!')
 })
 
-app.listen(port, () => {
-  console.log(`Chat app is listening at http://localhost:${port}`)
-})
\ No newline at end of file
+// Socket IO connection
+io.on('connection', (socket) => {
+  // Access session information
+  let cookieString = socket.request.headers.cookie;
+  let req = { headers : {cookie : cookieString} }
+  cookieSession({ keys: ['superdupersecret'] })(req, {}, function(){})
+
```

Comment:
Check that `req` has a `session` property before accessing. Same with `name`.

Also you can read the `name` value to a variable and use that instead of accessing nested objects.

Reply:
I approached this one by writing a method `name` that returns the user's name if `req` has a `session` property and `session` has a `name` property, but returns the string `Someone` if either of those properties don't exist. Take a look at the function in the next PR and let me know what you think.

Reply:
👍